spec/requests/posts_spec.rb

```diff
@@ -0,0 +1,45 @@
+require 'rails_helper'
+
+RSpec.describe 'Posts', type: :request do
+  describe 'If response status was correct.' do
+    it 'If response status was correct for user posts path' do
+      get user_posts_path(2)
+      expect(response).to have_http_status(200)
+    end
+  end
+
+  describe 'If response status was correct.' do
+    it 'If response status was correct for user post path and user ID' do
+      get user_post_path(1, 1)
+      expect(response).to have_http_status(200)
+    end
+  end
+
+  describe 'Render correct template' do
+    it 'renders the posts template' do
+      get user_posts_path(1)
+      expect(response).to render_template('posts/index')
+    end
+  end
+
+  describe 'Render correct template' do
+    it 'renders the show template' do
+      get user_post_path(1, 1)
+      expect(response).to render_template('posts/show')
+    end
+  end
+
+  describe 'Response body includes correct placeholder text' do
+    it 'displays the correct placeholder text' do
+      get user_posts_path(1)
+      expect(response.body).to include('Here is a list of posts')
+    end
+  end
+
+  describe 'Response body includes correct placeholder text' do
+    it 'displays the correct placeholder text' do
+      get user_post_path(1, 1)
+      expect(response.body).to include('Here is a list of post for a given user')
+    end
+  end
+end
```

Comment:
Please some of your tests are not passing, kindly fix this 👍🏽 
![Screenshot from 2023-05-13 01-01-36](https://github.com/huzaifakhan0308/blogsVault/assets/44624138/4f3a4b3e-a80b-45e1-baac-6f5be4ba373c)

Reply:
as you can see here, in my local machine, all tests are passing.

![Screenshot 2023-05-13 124523](https://github.com/huzaifakhan0308/blogsVault/assets/114409312/fac35a95-1517-4872-838a-363b0b40d14c)